app/src/main/java/mozilla/lockbox/presenter/CreateItemPresenter.kt

```diff
@@ -52,24 +52,26 @@ class CreateItemPresenter(
         return if (hasChanges) {
             listOf(
                 ItemDetailAction.CreateItemSaveChanges,
-                ToastNotificationAction.ShowSuccessfulCreateToast
+                ToastNotificationAction.ShowSuccessfulCreateToast,
+                ItemDetailAction.EndCreateItemSession
             )
         } else {
             listOf(ItemDetailAction.EndCreateItemSession)
         }
     }
 
-    override fun dismissChangesAction(hasChanges: Boolean): Action {
-        return if (hasChanges) {
-            DialogAction.DiscardChangesCreateDialog
+    override fun dismissChangesAction(hasChanges: Boolean): List<Action> =
+        if (hasChanges) {
+            listOf(DialogAction.DiscardChangesCreateDialog)
         } else {
-            ItemDetailAction.EndCreateItemSession
+            listOf(ItemDetailAction.EndCreateItemSession, RouteAction.ItemList)
         }
-    }
 
     override fun endEditingActions(): List<Action> {
+        val list = itemDetailStore.findSavedItem().blockingIterable().iterator().next()
         return listOf(
-            RouteAction.ItemList
+            RouteAction.ItemList,
+            RouteAction.DisplayItem(list.first().id)
```

Comment:
I think I'd prefer this method to be of the sort:

```kotlin
override fun endEditingActions(): Observable<List<Action>>
```

and then in the `ItemMutationPresenter` use a combination of 

```kotlin
.flatMap { endEditingActions() }
.flatMapIterable { it }
```

Overtime, I think these action methods may end up all being `Observable<List<Action>>`, but right now let's see how it works.